device/cuda/include/traccc/cuda/seeding/seeding_algorithm.hpp

```diff
@@ -54,11 +54,17 @@ class seeding_algorithm : public algorithm<seed_collection_types::buffer(
                                spacepoints_view) const override;
 
     private:
-    /// Sub-algorithm performing the spacepoint binning
-    spacepoint_binning m_spacepoint_binning;
-    /// Sub-algorithm performing the seed finding
-    seed_finding m_seed_finding;
+    /// Config objects
+    seedfinder_config m_finder_config;
+    spacepoint_grid_config m_grid_config;
+    seedfilter_config m_filter_config;
 
+    /// Memory resource used by the algorithm
+    traccc::memory_resource m_mr;
+    /// The copy object to use
+    vecmem::copy& m_copy;
+    /// The CUDA stream to use
+    stream& m_stream;
```

Comment:
The same here. Instead of adding unnecessary members to this class, let's just make sure that user-supplied "config objects" can be given to the algorithm's constructor. And then passed on directly to the underlying algorithms.

Reply:
Are you talking about putting the memory resources and copy objects and streams into the config objects? Because we should absolutely **not** be doing that. :worried:

Reply:
No, I'm not.

Currently all `seeding_algorithm` classes have 2 members: A "sub-algorithm" doing the spacepoint binning, and another "sub-algorithm" doing the seed finding.

The main algorithm's **only** purpose is to call those 2 sub-algorithms in sequence.

The configurations, memory resources, etc. are **only** needed by the sub-algorithms. The main algorithm doesn't use these directly ever.

So the setup in the current code is that the constructor or `seeding_algorithm` passes all the appropriate values to the constructors of the sub-algorithms, and then just holds on to the sub-algorithms. Not all the parameters that the sub-algorithms use internally.

Very long story short, in none of the `seeding_algorithm` classes should the list of members change in this MR. We should just extend the `seeding_algorithm(...)` constructors to be able to receive user-defined config objects. Along with the user-defined streams, memory resources, etc.

Reply:
Ok good.